### PR TITLE
Added Static build for Advanced augmentation operrations

### PR DESCRIPTION
**Pull request by yashasps**

### CODE REVIEW

File: src/modules/hip/hip_declarations.hpp [hunk 1]
```diff
@@ -12,6 +12,72 @@ void get_kernel_name(std::string &kernel_name, const RPPTensorFunctionMetaData &
...
+RppStatus
+color_cast_hip_batch_tensor_fp16(Rpp16f *srcPtr, Rpp16f *dstPtr, rpp::Handle &handle, RPPTensorFunctionMetaData &tensor_info);
+RppStatus
+color_cast_hip_batch_tensor_fp32(Rpp32f *srcPtr, Rpp32f *dstPtr, rpp::Handle &handle, RPPTensorFunctionMetaData &tensor_info);
+RppStatus
+color_cast_hip_batch_tensor_int8(Rpp8s *srcPtr, Rpp8s *dstPtr, rpp::Handle &handle, RPPTensorFunctionMetaData &tensor_info);
+
+RppStatus
+lut_hip_batch_tensor(Rpp8u *srcPtr, Rpp8u *dstPtr, Rpp8u *lut,rpp::Handle &handle, RPPTensorFunctionMetaData &tensor_info);
+// RppStatus
```
Comment (r-abishek) [inline src/modules/hip/hip_declarations.hpp:55]: Remove these 2 commented headers

File: src/modules/hip/hip_declarations.hpp [hunk 2]
```diff
@@ -12,6 +12,72 @@ void get_kernel_name(std::string &kernel_name, const RPPTensorFunctionMetaData &
...
+
+RppStatus
+non_linear_blend_hip_batch_tensor(Rpp8u *srcPtr1, Rpp8u *srcPtr2, Rpp8u *dstPtr, rpp::Handle &handle, RPPTensorFunctionMetaData &tensor_info);
+RppStatus
+non_linear_blend_hip_batch_tensor_fp16(Rpp16f *srcPtr1, Rpp16f *srcPtr2, Rpp16f *dstPtr, rpp::Handle &handle, RPPTensorFunctionMetaData &tensor_info);
+RppStatus
+non_linear_blend_hip_batch_tensor_fp32(Rpp32f *srcPtr1, Rpp32f *srcPtr2, Rpp32f *dstPtr, rpp::Handle &handle, RPPTensorFunctionMetaData &tensor_info);
+RppStatus
+non_linear_blend_hip_batch_tensor_int8(Rpp8s *srcPtr1, Rpp8s *srcPtr2, Rpp8s *dstPtr, rpp::Handle &handle, RPPTensorFunctionMetaData &tensor_info);
+
```
Comment (r-abishek) [inline src/modules/hip/hip_declarations.hpp:25]: Remove blank lines between functions in Line 25, 34, 43, 52, 61, 70.

File: src/modules/hip/hip_advanced_augmentation.cpp [hunk 3]
```diff
@@ -0,0 +1,924 @@
...
+        hip_exec_color_cast_batch_int8(srcPtr, dstPtr, handle, tensor_info, in_plnpkdind, out_plnpkdind, max_height, max_width);
+
+    #endif
+
+    return RPP_SUCCESS;
+}
+/******************** lut ********************/
+
+// RppStatus
+// lut_hip_batch(Rpp8u* srcPtr, Rpp8u* dstPtr, Rpp8u* lut, rpp::Handle &handle, RPPTensorFunctionMetaData &tensor_info)
```
Comment (r-abishek) [inline src/modules/hip/hip_advanced_augmentation.cpp:541]: Remove the commented lut_hip_batch function, Also remove the 6 other *_hip_batch functions. Let's just have the *_hip_batch_tensor for these 7 functions.

File: src/modules/hip/hip_advanced_augmentation.cpp [hunk 4]
```diff
@@ -0,0 +1,924 @@
...
+
+    #elif defined(STATIC)
+
+            hip_exec_lut_batch_tensor(srcPtr, dstPtr, lut, handle, tensor_info, in_plnpkdind, out_plnpkdind, max_height, max_width);
+    #endif
+    return RPP_SUCCESS;
+}
+
+RppStatus
+lut_hip_batch_tensor_fp16(Rpp16f *srcPtr1, Rpp16f *srcPtr2, Rpp16f *dstPtr, Rpp8u* lut, rpp::Handle &handle, RPPTensorFunctionMetaData &tensor_info)
```
Comment (r-abishek) [inline src/modules/hip/hip_advanced_augmentation.cpp:606]: Just for lut, remove this lut_hip_batch_tensor_fp16 function

File: src/modules/hip/hip_advanced_augmentation.cpp [hunk 5]
```diff
@@ -0,0 +1,924 @@
...
+//     max_size(handle_obj->mem.mgpu.csrcSize.height, handle_obj->mem.mgpu.csrcSize.width, handle.GetBatchSize(), &max_height, &max_width);
+// #if defined (HIPRTC)
+// #elif defined(STATIC)
+//      hip_exec_lut_batch_tensor_fp16(srcPtr1, srcPtr2, dstPtr, handle, in_plnpkdind, out_plnpkdind, max_height, max_width);
+// #endif
+    return RPP_SUCCESS;
+}
+
+RppStatus
+lut_hip_batch_tensor_fp32(Rpp32f *srcPtr, Rpp32f *dstPtr, Rpp32f* lut, rpp::Handle &handle, RPPTensorFunctionMetaData &tensor_info)
```
Comment (r-abishek) [inline src/modules/hip/hip_advanced_augmentation.cpp:620]: Just for lut, remove this lut_hip_batch_tensor_fp32 function too

File: src/modules/hip/hip_advanced_augmentation.cpp [hunk 6]
```diff
@@ -0,0 +1,924 @@
...
+#elif defined(STATIC)
+    hip_exec_non_linear_blend_batch(srcPtr1, srcPtr2, dstPtr, handle, tensor_info, in_plnpkdind, out_plnpkdind, max_height, max_width);
+#endif
+    return RPP_SUCCESS;
+}
+
+RppStatus
+non_linear_blend_hip_batch_tensor_fp16(Rpp16f *srcPtr1, Rpp16f *srcPtr2, Rpp16f *dstPtr, rpp::Handle &handle, RPPTensorFunctionMetaData &tensor_info)
+{
+//     int in_plnpkdind = getplnpkdind(tensor_info._in_format), out_plnpkdind = getplnpkdind(tensor_info._out_format);
```
Comment (r-abishek) [inline src/modules/hip/hip_advanced_augmentation.cpp:84]: For all the fp16 functions (7-1=6 funcs), keep the contents of the function uncommented here in the higher level API. Let's just comment out the contents of the kernel itself.

File: src/modules/hip/kernel/rpp_hip_host_decls.hpp [hunk 7]
```diff
@@ -7,6 +7,41 @@
...
+RppStatus hip_exec_color_cast_batch_fp16(Rpp16f *srcPtr, Rpp16f *dstPtr, rpp::Handle& handle, RPPTensorFunctionMetaData &tensor_info, Rpp32s in_plnpkdind, Rpp32s out_plnpkdind, Rpp32u max_height, Rpp32u max_width);
+RppStatus hip_exec_color_cast_batch_fp32(Rpp32f *srcPtr, Rpp32f *dstPtr, rpp::Handle& handle, RPPTensorFunctionMetaData &tensor_info, Rpp32s in_plnpkdind, Rpp32s out_plnpkdind, Rpp32u max_height, Rpp32u max_width);
+RppStatus hip_exec_color_cast_batch_int8(Rpp8s *srcPtr, Rpp8s *dstPtr, rpp::Handle& handle, RPPTensorFunctionMetaData &tensor_info, Rpp32s in_plnpkdind, Rpp32s out_plnpkdind, Rpp32u max_height, Rpp32u max_width);
+
+RppStatus hip_exec_lut_batch_tensor(Rpp8u *srcPtr, Rpp8u *dstPtr, Rpp8u* lut, rpp::Handle& handle, RPPTensorFunctionMetaData &tensor_info, Rpp32s in_plnpkdind, Rpp32s out_plnpkdind, Rpp32u max_height, Rpp32u max_width);
+// RppStatus hip_exec_lut_batch_fp16(Rpp16f *srcPtr, Rpp16f *dstPtr, Rpp16f* lut, rpp::Handle& handle, RPPTensorFunctionMetaData &tensor_info, Rpp32s in_plnpkdind, Rpp32s out_plnpkdind, Rpp32u max_height, Rpp32u max_width);
```
Comment (r-abishek) [inline src/modules/hip/kernel/rpp_hip_host_decls.hpp:31]: Remove the commented fp16, fp32 versions of lut too

File: src/modules/hip/kernel/rpp_hip_host_decls.hpp [hunk 8]
```diff
@@ -7,6 +7,41 @@
...
+// RppStatus hip_exec_lut_batch_fp32(Rpp32f *srcPtr, Rpp32f *dstPtr, Rpp32f* lut, rpp::Handle& handle, RPPTensorFunctionMetaData &tensor_info, Rpp32s in_plnpkdind, Rpp32s out_plnpkdind, Rpp32u max_height, Rpp32u max_width);
+RppStatus hip_exec_lut_batch_int8(Rpp8s *srcPtr, Rpp8s *dstPtr, Rpp8s* lut, rpp::Handle& handle, RPPTensorFunctionMetaData &tensor_info, Rpp32s in_plnpkdind, Rpp32s out_plnpkdind, Rpp32u max_height, Rpp32u max_width);
```
Comment (r-abishek) [inline src/modules/hip/kernel/rpp_hip_host_decls.hpp:33]: Change this name to hip_exec_lut_batch_tensor_int8

File: src/modules/hip/kernel/rpp_hip_host_decls.hpp [hunk 9]
```diff
@@ -7,6 +7,41 @@
...
+
+RppStatus hip_exec_color_cast_batch(Rpp8u *srcPtr, Rpp8u *dstPtr, rpp::Handle& handle, RPPTensorFunctionMetaData &tensor_info, Rpp32s in_plnpkdind, Rpp32s out_plnpkdind, Rpp32u max_height, Rpp32u max_width);
+RppStatus hip_exec_color_cast_batch_fp16(Rpp16f *srcPtr, Rpp16f *dstPtr, rpp::Handle& handle, RPPTensorFunctionMetaData &tensor_info, Rpp32s in_plnpkdind, Rpp32s out_plnpkdind, Rpp32u max_height, Rpp32u max_width);
+RppStatus hip_exec_color_cast_batch_fp32(Rpp32f *srcPtr, Rpp32f *dstPtr, rpp::Handle& handle, RPPTensorFunctionMetaData &tensor_info, Rpp32s in_plnpkdind, Rpp32s out_plnpkdind, Rpp32u max_height, Rpp32u max_width);
+RppStatus hip_exec_color_cast_batch_int8(Rpp8s *srcPtr, Rpp8s *dstPtr, rpp::Handle& handle, RPPTensorFunctionMetaData &tensor_info, Rpp32s in_plnpkdind, Rpp32s out_plnpkdind, Rpp32u max_height, Rpp32u max_width);
+
+RppStatus hip_exec_lut_batch_tensor(Rpp8u *srcPtr, Rpp8u *dstPtr, Rpp8u* lut, rpp::Handle& handle, RPPTensorFunctionMetaData &tensor_info, Rpp32s in_plnpkdind, Rpp32s out_plnpkdind, Rpp32u max_height, Rpp32u max_width);
```
Comment (r-abishek) [inline src/modules/hip/kernel/rpp_hip_host_decls.hpp:30]: Change this name to just hip_exec_lut_batch to match others

File: src/modules/hip/kernel/rpp_hip_host_decls.hpp [hunk 10]
```diff
@@ -7,6 +7,41 @@
 
 // advanced_augmentations
 
+RppStatus hip_exec_non_linear_blend_batch(Rpp8u *srcPtr1, Rpp8u *srcPtr2, Rpp8u *dstPtr, rpp::Handle& handle, RPPTensorFunctionMetaData &tensor_info, Rpp32s in_plnpkdind, Rpp32s out_plnpkdind, Rpp32u max_height, Rpp32u max_width);
+RppStatus hip_exec_non_linear_blend_batch_fp16(Rpp16f *srcPtr1, Rpp16f *srcPtr2, Rpp16f *dstPtr, rpp::Handle& handle, RPPTensorFunctionMetaData &tensor_info, Rpp32s in_plnpkdind, Rpp32s out_plnpkdind, Rpp32u max_height, Rpp32u max_width);
+RppStatus hip_exec_non_linear_blend_batch_fp32(Rpp32f *srcPtr1, Rpp32f *srcPtr2, Rpp32f *dstPtr, rpp::Handle& handle, RPPTensorFunctionMetaData &tensor_info, Rpp32s in_plnpkdind, Rpp32s out_plnpkdind, Rpp32u max_height, Rpp32u max_width);
+RppStatus hip_exec_non_linear_blend_batch_int8(Rpp8s *srcPtr1, Rpp8s *srcPtr2, Rpp8s *dstPtr, rpp::Handle& handle, RPPTensorFunctionMetaData &tensor_info, Rpp32s in_plnpkdind, Rpp32s out_plnpkdind, Rpp32u max_height, Rpp32u max_width);
+
```
Comment (r-abishek) [inline src/modules/hip/kernel/rpp_hip_host_decls.hpp:14]: Remove gap lines in 14, 19, 24, 29, 34, 39

File: src/modules/hip/kernel/look_up_table.cpp [hunk 11]
```diff
@@ -1,127 +1,224 @@
...
+        output[out_pix_index] = lutPtr[lutIndex];
+        in_pix_index += inc[id_z];
+        out_pix_index += dst_inc[id_z];
+        }
     }
-  }
 }
+
+#if defined(STATIC)
+RppStatus hip_exec_lut_batch_tensor(Rpp8u *srcPtr, Rpp8u *dstPtr, Rpp8u* lut, rpp::Handle& handle, RPPTensorFunctionMetaData &tensor_info, Rpp32s in_plnpkdind, Rpp32s out_plnpkdind, Rpp32u max_height, Rpp32u max_width)
```
Comment (r-abishek) [inline src/modules/hip/kernel/look_up_table.cpp:163]: Change this to hip_exec_lut_batch() too

File: src/modules/hip/kernel/look_up_table.cpp [hunk 12]
```diff
@@ -1,127 +1,224 @@
...
-        batch_index[id_z] + (id_x + id_y * max_width[id_z]) * in_pln_pkd_ind;
-    long out_pix_index =
-        batch_index[id_z] + (id_x + id_y * max_width[id_z]) * out_pln_pkd_ind;
-    int luptrIndex = id_z << 8;
-    for (int indextmp = 0; indextmp < channel; indextmp++) {
-      int lutIndex = luptrIndex + input[in_pix_index];
-      output[out_pix_index] = lutPtr[lutIndex];
-      in_pix_index += inc[id_z];
-      out_pix_index += dst_inc[id_z];
+extern "C" __global__ void look_up_table_batch_tensor(unsigned char *input,
```
Comment (r-abishek) [inline src/modules/hip/kernel/look_up_table.cpp:97]: I don't think the 'tensor' word appears in the lower level files. Lets rename:
- Kernel name @L97: look_up_table_batch_tensor -> lut_batch
- Kernel name @L129: look_up_table_batch_tensor_int8 -> lut_batch_int8
- Kernel call @L173: look_up_table_batch_tensor -> lut_batch
- Kernel call @L204: look_up_table_batch_tensor_int8 -> lut_batch_int8
Try for build issues.

File: src/modules/hip/kernel/look_up_table.cpp [hunk 13]
```diff
@@ -1,127 +1,224 @@
...
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if (id_x < width[id_z] && id_y < height[id_z])
+    {
+        long in_pix_index = batch_index[id_z] + (id_x + id_y * max_width[id_z]) * in_pln_pkd_ind;
+        long out_pix_index = batch_index[id_z] + (id_x + id_y * max_width[id_z]) * out_pln_pkd_ind;
+        int luptrIndex = id_z << 8;
+        for (int indextmp = 0; indextmp < channel; indextmp++)
+        {
+        int lutIndex = luptrIndex + input[in_pix_index];
```
Comment (r-abishek) [inline src/modules/hip/kernel/look_up_table.cpp:121]: fix indentation

File: src/modules/hip/kernel/look_up_table.cpp [hunk 14]
```diff
@@ -1,127 +1,224 @@
...
+
+    if (id_x < width[id_z] && id_y < height[id_z])
+    {
+        long in_pix_index = batch_index[id_z] + (id_x + id_y * max_width[id_z]) * in_pln_pkd_ind;
+        long out_pix_index = batch_index[id_z] + (id_x + id_y * max_width[id_z]) * out_pln_pkd_ind;
+        int luptrIndex = id_z << 8;
+
+        for (int indextmp = 0; indextmp < channel; indextmp++)
+        {
+        int lutIndex = luptrIndex + input[in_pix_index] + 128;
```
Comment (r-abishek) [inline src/modules/hip/kernel/look_up_table.cpp:154]: fix indentation

File: src/modules/rppi_advanced_augmentations.cpp [hunk 15]
```diff
@@ -23,2481 +23,2368 @@ using namespace std::chrono;
...
-				rpp::deref(rppHandle),
-				tensor_info);
-		}
-	}
+    {
+        if (in_tensor_type == RPPTensorDataType::U8)
+        {
+            lut_hip_batch_tensor(static_cast<Rpp8u *>(srcPtr), static_cast<Rpp8u *>(dstPtr), static_cast<Rpp8u *>(lut), rpp::deref(rppHandle), tensor_info);
+        }
+        // else if (in_tensor_type == RPPTensorDataType::FP16)
```
Comment (r-abishek) [inline src/modules/rppi_advanced_augmentations.cpp:1388]: Remove FP16, FP32 conditions commented lines

**Review by r-abishek — CHANGES_REQUESTED** [review]
Needs a few changes as commented, ensure unittests run fine after the changes.

File: src/modules/hip/kernel/look_up_table.cpp (hunk 12 above)
Comment (yashasps) [reply src/modules/hip/kernel/look_up_table.cpp:97]: Abishek, I tried to do it, but it  is throwing an """ conflicting types for 'look_up_table_batch' """

File: src/modules/hip/kernel/look_up_table.cpp (hunk 11 above)
Comment (yashasps) [reply src/modules/hip/kernel/look_up_table.cpp:163]: done

File: src/modules/hip/kernel/rpp_hip_host_decls.hpp (hunk 10 above)
Comment (yashasps) [reply src/modules/hip/kernel/rpp_hip_host_decls.hpp:14]: done

File: src/modules/hip/kernel/rpp_hip_host_decls.hpp (hunk 9 above)
Comment (yashasps) [reply src/modules/hip/kernel/rpp_hip_host_decls.hpp:30]: done

File: src/modules/hip/kernel/rpp_hip_host_decls.hpp (hunk 7 above)
Comment (yashasps) [reply src/modules/hip/kernel/rpp_hip_host_decls.hpp:31]: done

File: src/modules/hip/hip_advanced_augmentation.cpp (hunk 6 above)
Comment (yashasps) [reply src/modules/hip/hip_advanced_augmentation.cpp:84]: done

File: src/modules/hip/hip_advanced_augmentation.cpp (hunk 5 above)
Comment (yashasps) [reply src/modules/hip/hip_advanced_augmentation.cpp:620]: done

File: src/modules/hip/hip_advanced_augmentation.cpp (hunk 4 above)
Comment (yashasps) [reply src/modules/hip/hip_advanced_augmentation.cpp:606]: done

File: src/modules/hip/hip_advanced_augmentation.cpp (hunk 3 above)
Comment (yashasps) [reply src/modules/hip/hip_advanced_augmentation.cpp:541]: done

File: src/modules/hip/hip_declarations.hpp (hunk 2 above)
Comment (yashasps) [reply src/modules/hip/hip_declarations.hpp:25]: done

File: src/modules/hip/hip_declarations.hpp (hunk 1 above)
Comment (yashasps) [reply src/modules/hip/hip_declarations.hpp:55]: done

File: src/modules/hip/kernel/look_up_table.cpp (hunk 13 above)
Comment (yashasps) [reply src/modules/hip/kernel/look_up_table.cpp:121]: done

File: src/modules/hip/kernel/look_up_table.cpp (hunk 14 above)
Comment (yashasps) [reply src/modules/hip/kernel/look_up_table.cpp:154]: done

File: src/modules/rppi_advanced_augmentations.cpp (hunk 15 above)
Comment (yashasps) [reply src/modules/rppi_advanced_augmentations.cpp:1388]: done

**Review by r-abishek — APPROVED** [review]

**Pull request merged**